lib/ansible/modules/windows/win_dns_record.ps1

```diff
@@ -0,0 +1,148 @@
+#!powershell
+
+# Copyright: (c) 2019, Hitachi ID Systems, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
```

Comment:
This module is still using the legacy Powershell module interface.
We do have a new Powershell module interface which has many improvements.
Would you care to rewrite this module using the new interface?
Please look at some other Windows modules for examples.

Reply:
This was written by following the [developer guide](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general_windows.html#windows-new-module-development), which still recommends the legacy module. My bias is to always studiously avoid stepping outside the documented box -- meaning I would prefer waiting for the developer guide to show some new path before jumping in.

Reply:
You're correct to go by the docs however that page has been updated for Ansible 2.8 and you can view it [here](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_general_windows.html#windows-new-module-development).

The new module interface was introduced in the latest developmental release and would be great if you could use this instead. You wouldn't have to change a lot of code to use it.

Reply:
Tried to convert it in https://github.com/johnboy2/ansible/pull/2/commits/6a7d92a0791e5d58d11cd410d3a4c45d549820b7 . Tests are OK after conversion, but don know how to:
* use `prepared` in `diff`
* use `required_if` in this context

Reply:
I have this solved in https://github.com/johnboy2/ansible/pull/2 
Seems to me that `required_if` don't apply in this case.
Tests are OK